pkg/synthetictests/networking.go

```diff
@@ -44,16 +45,30 @@ func testPodSandboxCreation(events monitorapi.Intervals) []*ginkgo.JUnitTestCase
 		}
 		deletionTime := getPodDeletionTime(eventsForPods[event.Locator], event.Locator)
 		if deletionTime == nil {
+			// mark sandboxes errors as flakes if networking is being updated
+			// these pods eventually get created
+			operatorsProgressing := intervalcreation.IntervalsFromEvents_OperatorProgressing(events, event.From, event.To)
```

Comment:
Hrm, this could be O(N^2) on a pretty big N, have you verified that IntervalsFromEvents uses binary search?

Reply:
Seems `IntervalsFromEvents_OperatorProgressing` is `O(N)`

Reply:
The list should be sorted, so if you know from to you can do a binary search o(logn) to find the start and then same for the end.  Or maybe just at the beginning do a single pass and calculate all the intervals that the operator is progressing (which should be very small O) and then just do the smaller loop here?

Reply:
(see intervals.go / monitor.go for a method that already uses sort.Search() to do this)

Reply:
Reworked this to use monitorapi functions:
* `CopyAndSort` to create a copy of events and sort them by type
* `IntervalsFromEvents_OperatorProgressing` to build a list of operator progressing events
* `sort.Search` to find events for network/machine-config